### DISCUSSION TITLE
Reversing Animations?

### DISCUSSION THREAD

**Discussion by waffle-lord** [AvaloniaUI/Avalonia#12149] [Q&A] [answered]
Is it possible to automatically reverse an animation applied through a style?

I want to animate a BoxShadow on pointer over, which works fine, but there doesn't appear to be any way to animate the style when the pointer leaves the control.

What would be the best way to animate out this shadow when the mouse leave the control?

Code:
```axaml
<Style Selector="Border:pointerover">
       <Setter Property="BoxShadow" Value="2 2 10 1 black"/>
	<Style.Animations>
		<Animation Duration="0:0:0.1">
			<KeyFrame Cue="0%">
				<Setter Property="BoxShadow" Value="0 0 0 0 black"/>
			</KeyFrame>
			<KeyFrame Cue="100%">
				<Setter Property="BoxShadow" Value="2 2 10 1 black"/>
			</KeyFrame>
		</Animation>
	</Style.Animations>
</Style>
```
Which looks like this
![blah](https://github.com/AvaloniaUI/Avalonia/assets/76401815/5b367e9c-4b64-4970-9cd3-0cc3460609c4)

**Comment by maxkatz6** [comment]
> What would be the best way to animate out this shadow when the mouse leave the control?

You can add another animation for the normal state, i.e. without :pointerover

**Reply by lukaszossowski** [reply]
Have you considered transitions? If you add a transition in the main style all changes to related property will be animated automatically.

**Reply by waffle-lord** [reply]
Adding a normal style with an animation doesn't seem to have any affect on the control when the pointer leaves it.

I also tried using the `BoxShadowsTransition` but I don't know what the property value should be. There isn't much (any?) information on how to use this transition that I could find  :(

Code in case I'm doing it wrong:
```axaml
<Style Selector="Border:pointerover">
			<Setter Property="BoxShadow" Value="2 2 10 1 black"/>
			<!--<Setter Property="Transitions">
				<Transitions>
					<BoxShadowsTransition Property="BoxShadow" Duration="0:0:0.2"/>
				</Transitions>
			</Setter>-->
			<Style.Animations>
				<Animation Duration="0:0:0.2">
					<KeyFrame Cue="0%">
						<Setter Property="BoxShadow" Value="0 0 0 0 black"/>
					</KeyFrame>
					<KeyFrame Cue="100%">
						<Setter Property="BoxShadow" Value="2 2 10 1 black"/>
					</KeyFrame>
				</Animation>
			</Style.Animations>
		</Style>
		<Style Selector="Border">
			<Setter Property="BoxShadow" Value="0 0 0 0 black"/>
			<Style.Animations>
				<Animation Duration="0:0:0.2">
					<KeyFrame Cue="0%">
						<Setter Property="BoxShadow" Value="2 2 10 1 black"/>
					</KeyFrame>
					<KeyFrame Cue="100%">
						<Setter Property="BoxShadow" Value="0 0 0 0 black"/>
					</KeyFrame>
				</Animation>
			</Style.Animations>
		</Style>
```

**Comment by waffle-lord** [comment, marked as answer]
I was able to get it working with transitions.

Here are the styles I had to use

```axaml
<Style Selector="Border">
	<Setter Property="BoxShadow" Value="0 0 0 0 black"/>
	<Setter Property="Transitions">
		<Transitions>
			<BoxShadowsTransition Property="BoxShadow" Duration="0:0:0.2"/>
		</Transitions>
	</Setter>
</Style>

<Style Selector="Border:pointerover">
	<Setter Property="BoxShadow" Value="2 2 5 2 black"/>
</Style>
```
